fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountReadPlatformServiceImpl.java

```diff
@@ -230,6 +246,495 @@ public SavingsAccountData retrieveOne(final Long accountId) {
+                    }
+
+                    savingsAccountData.setSavingsAccountTransactionData(savingsAccountTransactionData);
+                }
+
+                if (chargeId != null && !chargeDetails.containsValue(chargeId)) {
+                    final boolean isPenalty = rs.getBoolean("isPenaltyCharge");
+                    final BigDecimal chargeAmount = JdbcSupport.getBigDecimalDefaultToNullIfZero(rs, "chargeAmount");
+                    final Integer chargesTimeType = rs.getInt("chargeTimeType");
+                    final EnumOptionData enumOptionDataForChargesTimeType = new EnumOptionData(chargesTimeType.longValue(), null, null);
+                    final SavingsAccountChargeData savingsAccountChargeData = new SavingsAccountChargeData(chargeId, chargeAmount,
+                            enumOptionDataForChargesTimeType, isPenalty);
+
+                    final Long chargesPaidById = rs.getLong("chargesPaidById");
+                    final BigDecimal chargesPaid = JdbcSupport.getBigDecimalDefaultToNullIfZero(rs, "paidByAmount");
+                    final SavingsAccountChargesPaidByData savingsAccountChargesPaidByData = new SavingsAccountChargesPaidByData(
+                            chargesPaidById, chargesPaid);
+                    savingsAccountChargesPaidByData.setSavingsAccountChargeData(savingsAccountChargeData);
+                    if (savingsAccountChargesPaidByData != null) {
+                        savingsAccountTransactionData.setChargesPaidByData(savingsAccountChargesPaidByData);
+                    }
+
+                    chargeDetails.put("id", chargeId);
+                }
+
+                if (taxDetailId != null && !taxDetails.containsValue(taxDetailId)) {
+                    final BigDecimal amount = JdbcSupport.getBigDecimalDefaultToNullIfZero(rs, "taxAmount");
+                    final BigDecimal percentage = JdbcSupport.getBigDecimalDefaultToNullIfZero(rs, "taxPercentage");
+                    final Long debitId = rs.getLong("debitAccountId");
+                    final Long creditId = rs.getLong("creditAccountId");
+                    final GLAccountData debitAccount = GLAccountData.createFrom(debitId);
+                    final GLAccountData creditAccount = GLAccountData.createFrom(creditId);
+
+                    if (taxComponentId != null) {
+                        final TaxComponentData taxComponent = TaxComponentData.createTaxComponent(taxComponentId, percentage, debitAccount,
+                                creditAccount);
+                        savingsAccountTransactionData.setTaxDetails(new TaxDetailsData(taxComponent, amount));
+                    }
+
+                    taxDetails.put("id", taxDetailId);
+                }
+
+            }
+            if (savingsAccountData != null) {
+                savingsAccountDataList.add(savingsAccountData);
+            }
+            return savingsAccountDataList;
+
+            // final String productName = rs.getString("productName");
```

Comment:
If we need to keep the native SQL approach, we shall probably fix the comments before merging.